src/test/clojure/core2/core/datalog_test.clj

```diff
@@ -1012,3 +1012,197 @@
+                                                                          [id :application_time_start app-start]
+                                                                          [id :application_time_end app-end]
+                                                                          [(= (- #inst "1970-01-08" #inst "1970-01-01")
+                                                                              (- app-end app-start))]]})
+                                                             (map (fn [{:keys [id app-start app-end]}]
+                                                                    [:delete id {:app-time-start #inst "1998-01-05" #_app-start ;; TODO #648
+                                                                                 :app-time-end #inst "1998-01-12" #_app-end}]))))}]
+                              [:call :delete-1-week-records]]
+                             {:sys-time #inst "1998-01-30"})
+
+          !tx7 (c2/submit-tx tu/*node*
+                             [[:put {:id 2 :customer-number 827 :property-number 3621} {:app-time-start #inst "1998-01-15"}]]
+                             {:sys-time #inst "1998-01-31"})]
+
+      (t/is (= [{:cust 145 :app-start (util/->zdt #inst "1998-01-10")}]
+               (q '{:find [cust app-start]
+                    :where [(for-all-app-time id)
+                            [id :application_time_start app-start]
+                            [id :customer-number cust]]}, !tx0, nil))
+            "as-of 14 Jan")
+
+      (t/is (= [{:cust 145, :app-start (util/->zdt #inst "1998-01-10")}
+                {:cust 827, :app-start (util/->zdt #inst "1998-01-15")}]
+               (q '{:find [cust app-start]
+                    :where [(for-all-app-time id)
+                            [id :application_time_start app-start]
+                            [id :customer-number cust]]}, !tx1, nil))
+            "as-of 18 Jan")
+
+      (t/is (= [{:cust 145, :app-start (util/->zdt #inst "1998-01-05")}
+                {:cust 827, :app-start (util/->zdt #inst "1998-01-12")}]
+               (q '{:find [cust app-start]
+                    :order-by [[app-start :asc]]
+                    :where [(for-all-app-time id)
+                            [id :application_time_start app-start]
+                            [id :customer-number cust]]}, !tx5, nil))
+            "as-of 29 Jan")
+
+      (t/is (= [{:cust 827, :app-start (util/->zdt #inst "1998-01-12"), :app-end (util/->zdt #inst "1998-01-20")}]
+               (q '{:find [cust app-start app-end]
+                    :order-by [[app-start :asc]]
+                    :where [(for-all-app-time id)
+                            [id :application_time_start app-start]
+                            [id :application_time_end app-end]
+                            [id :customer-number cust]]}, !tx6, nil))
+            "'as best known' (as-of 30 Jan)")
+
+      (t/is (= [{:prop 3621}]
```

Comment:
Note that this test only returns 1 result, which means there is some unresolved semantic drift with what I cooked up previously:
https://github.com/xtdb/core2/blob/31255227aa1b3c3e2d104da8a2b13310b7124503/src/test/resources/core2/sql/logic_test/direct-sql/sl-demo.test#L677-L693

The original from the textbook is (and presumably only returns one result but I'm not sure that can be confirmed):
```sql
What properties are owned by the customer who owns property
7797?
SELECT P2.property_number
FROM Prop_Owner AS P1, Prop_Owner AS P2
WHERE P1.property_number = 7797
AND P2.property_number <> P1.property_number
AND P1.customer_number = P2.customer_number
```

...I haven't dug into the logical plans for these, but I reckon the top-level semantics are sufficiently different that the right answer was probably to add a `distinct` in the sl-demo version in the first instance.